api/types/saml_idp_service_provider.go

```diff
@@ -35,6 +31,10 @@ type SAMLIdPServiceProvider interface {
 	GetEntityDescriptor() string
 	// SetEntityDescriptor sets the entity descriptor of the service provider.
 	SetEntityDescriptor(string)
+	// GetEntityID returns the entity ID.
+	GetEntityID() string
+	// SetEntityID sets the entity ID.
+	SetEntityID(string)
```

Comment:
Yes, this will be used by the SAML IdP (enterprise feature).

It should always be pulled from the descriptor, however in order to look it up we need to either have it pre-parsed (like this PR) or parse every single descriptor to find the entity ID when we're looking for it. This will save us from needing to do that parse every time.